### ISSUE TITLE
Leaflet 1.0.0-beta2 low performances when dragging the map in IE11

### ISSUE THREAD

**Issue by fnicollet** [Leaflet/Leaflet#3863]
Hello,

I am trying Leaflet 1.0.0-beta2 as it seems to solve performance issues in IE11 for leaflet-markercluster.
But I am noticing a lag in IE11 with just a tile layer, without even adding leaflet-markercluster. To reproduce the issue, just add a map with any tile layer and make sure it is 100% of the page, to get as many tiles to show up as possible (I try on a touch-enabled laptop in standard resolution (1366px wide)
As new tiles appear (with some kind of transition), the FPS drop. Here is the performance panel from IE while only panning a map:

![image](https://cloud.githubusercontent.com/assets/1240481/10031833/a46fc644-617f-11e5-85ce-90843ad62c14.png)

Having the IE dev tools opened with the profiler turned on makes the bug even more noticeable.

The same code with 0.7.x has no performance issue. I'm not sure where to look in the code, a lot has changed.

If you need more information to debug this, just tell me
Thanks,
Fabien

**Comment by mourner** [comment]
This sounds serious. Fabien, it would be immensely helpful if you could `git bisect` to find a culprit commit.

**Comment by fnicollet** [comment]
I'm at work at the moment but should be able to bisect it tonight, no
problem
Le 23 sept. 2015 10:17, "Vladimir Agafonkin" <notifications@github.com> a
écrit :

**Comment by fnicollet** [comment]
Alright, I tried to bisect the issue but first, I needed to find a commit that would work in leaflet-master.

I am using /debug/map/map-mobile.html for my test, which seem to be the simplest example.
So I went back in time and checkout the code from the 21st commit page:
https://github.com/Leaflet/Leaflet/commits?page=21

Commit ed06a52 from 19 Mar 2014 already has the lag issue in IE.

If I go back to page 22, the tiles don't load at that point, there was probably an issue in between that was fixed before ed06a52. 
Anyway, even if I get the "image not found icon" instead of the tile, I can still feel that the map lags when I pan around.
I tried going back until page 35 (which is like very old !) and at some point, there is not even any image nor image not found at all, the map is blank with no error, so I guess it is like too far.

Can someone confirm the lag issue in IE with a full screen map? Because it feels weird that it would have been broken for so long

Fabien

**Comment by fnicollet** [comment]
Looks like I might have been wrong all along, the performances drop as soon as you open the dev tools (profiling or not). With the dev tools closed, the FPS drop is barely noticeable.
Why would the FPS drop when the dev tools are opened? Leaflet doesn't even log anything.

**Comment by mourner** [comment]
@fnicollet so, IE doesn't lag with devtools open on 0.7.5, but lags with master? Huh, that's weird, but that's IE, you know. Maybe try bisecting again with that knowledge?

**Comment by fnicollet** [comment]
Yes, exactly. IE debugger can do weird things, I had issues before with source maps embedded in JS files, makes the dev tools freeze along with your application (I disabled the source-map for my tests)
It's hard to bisect because at some point, leaflet-master didn't work in IE at all, and the dev tools issue was already present months ago.

Anyway, i'll close this issue as this is not really an issue in usual release conditions.

Oh and while I have your attention, I have opened an issue on a compat branch for mapbox-gl-leaflet and leaflet-master:
https://github.com/mapbox/mapbox-gl-leaflet/issues/21
There is still a small issue to fix with zoom to make it work, if you have some time :)

Fabien
